client/network/sync/src/engine.rs

```diff
@@ -0,0 +1,929 @@
+						log::warn!(target: "sync", "response channel closed for `NumSyncPeers`");
+					}
+				},
+				ToServiceCommand::NumQueuedBlocks(tx) => {
+					if let Err(_) = tx.send(self.chain_sync.status().queued_blocks) {
+						log::warn!(target: "sync", "response channel closed for `NumQueuedBlocks`");
+					}
+				},
+				ToServiceCommand::NumDownloadedBlocks(tx) => {
+					if let Err(_) = tx.send(self.chain_sync.num_downloaded_blocks()) {
+						log::warn!(target: "sync", "response channel closed for `NumDownloadedBlocks`");
+					}
+				},
+				ToServiceCommand::NumSyncRequests(tx) => {
+					if let Err(_) = tx.send(self.chain_sync.num_sync_requests()) {
+						log::warn!(target: "sync", "response channel closed for `NumSyncRequests`");
+					}
+				},
+				ToServiceCommand::PeersInfo(tx) => {
+					let peers_info =
+						self.peers.iter().map(|(id, peer)| (*id, peer.info.clone())).collect();
+					if let Err(_) = tx.send(peers_info) {
+						log::warn!(target: "sync", "response channel closed for `PeersInfo`");
+					}
+				},
+				ToServiceCommand::OnBlockFinalized(hash, header) =>
+					self.chain_sync.on_block_finalized(&hash, *header.number()),
+			}
+		}
+
+		while let Poll::Ready(result) = self.chain_sync.poll(cx) {
+			self.process_block_announce_validation_result(result);
+		}
+
+		Poll::Pending
+	}
+
+	/// Called by peer when it is disconnecting.
+	///
+	/// Returns a result if the handshake of this peer was indeed accepted.
+	pub fn on_sync_peer_disconnected(&mut self, peer: PeerId) -> Result<(), ()> {
+		if self.important_peers.contains(&peer) {
+			log::warn!(target: "sync", "Reserved peer {} disconnected", peer);
+		} else {
+			log::debug!(target: "sync", "{} disconnected", peer);
+		}
+
+		if let Some(_peer_data) = self.peers.remove(&peer) {
+			self.chain_sync.peer_disconnected(&peer);
+			self.default_peers_set_no_slot_connected_peers.remove(&peer);
+			self.event_streams
+				.retain(|stream| stream.unbounded_send(SyncEvent::PeerDisconnected(peer)).is_ok());
+			Ok(())
+		} else {
+			Err(())
+		}
+	}
+
+	/// Called on the first connection between two peers on the default set, after their exchange
+	/// of handshake.
+	///
+	/// Returns `Ok` if the handshake is accepted and the peer added to the list of peers we sync
+	/// from.
+	pub fn on_sync_peer_connected(
+		&mut self,
+		who: PeerId,
+		status: BlockAnnouncesHandshake<B>,
+	) -> Result<(), ()> {
+		log::trace!(target: "sync", "New peer {} {:?}", who, status);
+
+		if self.peers.contains_key(&who) {
+			log::error!(target: "sync", "Called on_sync_peer_connected with already connected peer {}", who);
+			debug_assert!(false);
+			return Err(())
+		}
+
+		if status.genesis_hash != self.genesis_hash {
+			log::log!(
+				target: "sync",
+				if self.important_peers.contains(&who) { log::Level::Warn } else { log::Level::Debug },
+				"Peer is on different chain (our genesis: {} theirs: {})",
+				self.genesis_hash, status.genesis_hash
+			);
```

Comment:
I know that this was just copied. However, as we have for `disconnect` also the extra message for reserved nodes we should do the same here! Basically what we do below for boot nodes should be printed for reserved nodes as well.

Reply:
Then this here is just a debug message.